### PR TITLE
New documentation with Jupyter book

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

Documentation with Jupyter Book. It can be found here: https://recommenders-team.github.io/recommenders

### Related Issues

#2016

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Comment by miguelgfierro** [comment]
If I add a relative path, it doesn't show in the table of contents
```
parts:
  - caption: Getting Started
    chapters:
      - file: ../SETUP.md
```

**Comment by miguelgfierro** [comment]
Current errors:
```
WARNING: autodoc: failed to import module 'cosmos_cli' from module 'recommenders.datasets'; the following exception was raised:
No module named 'pydocumentdb'
/home/u/MS/recommenders/recommenders/evaluation/spark_evaluation.py:docstring of recommenders.evaluation.spark_evaluation.SparkRatingEvaluation.rsquared:5: WARNING: Duplicate explicit target name: "here".
/home/u/MS/recommenders/recommenders/evaluation/spark_evaluation.py:docstring of recommenders.evaluation.spark_evaluation.SparkRatingEvaluation.rsquared:5: WARNING: Duplicate explicit target name: "here".
/home/u/MS/recommenders/recommenders/evaluation/spark_evaluation.py:docstring of recommenders.evaluation.spark_evaluation.SparkRatingEvaluation.rsquared:5: WARNING: Duplicate explicit target name: "here".
/home/u/MS/recommenders/recommenders/evaluation/spark_evaluation.py:docstring of recommenders.evaluation.spark_evaluation.SparkRatingEvaluation.rsquared:5: WARNING: Duplicate explicit target name: "here".
/home/u/MS/recommenders/docs/index.rst:10: WARNING: toctree directive not expected with external-toc [etoc.toctree]
/home/u/MS/recommenders/docs/intro.md:: WARNING: toctree contains reference to nonexisting document '../SETUP' [etoc.ref]
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4ALSTMCell.call:11: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4ALSTMCell.call:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4ALSTMCell.call:31: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4ALSTMCell.call:32: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4LSTMCell.call:11: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4LSTMCell.call:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4LSTMCell.call:31: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/deeprec/models/sequential/rnn_cell_implement.py:docstring of recommenders.models.deeprec.models.sequential.rnn_cell_implement.Time4LSTMCell.call:32: WARNING: Block quote ends without a blank line; unexpected unindent.
WARNING: autodoc: failed to import module 'geoimc_algorithm' from module 'recommenders.models.geoimc'; the following exception was raised:
No module named 'pymanopt'
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.ComputeMasking.call:11: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.ComputeMasking.call:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.ComputeMasking.call:31: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.ComputeMasking.call:32: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.OverwriteMasking.call:11: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.OverwriteMasking.call:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.OverwriteMasking.call:31: ERROR: Unexpected indentation.
/home/u/MS/recommenders/recommenders/models/newsrec/models/layers.py:docstring of recommenders.models.newsrec.models.layers.OverwriteMasking.call:32: WARNING: Block quote ends without a blank line; unexpected unindent.
WARNING: autodoc: failed to import module 'RLRMCalgorithm' from module 'recommenders.models.rlrmc'; the following exception was raised:
No module named 'pymanopt'
WARNING: autodoc: failed to import module 'conjugate_gradient_ms' from module 'recommenders.models.rlrmc'; the following exception was raised:
No module named 'pymanopt'
/home/u/MS/recommenders/recommenders/models/sar/sar_singlenode.py:docstring of recommenders.models.sar.sar_singlenode.SARSingleNode.fit:3: ERROR: Content block expected for the "note" directive; none found.
/home/u/MS/recommenders/recommenders/utils/notebook_utils.py:docstring of recommenders.utils.notebook_utils.execute_notebook:3: ERROR: Content block expected for the "note" directive; none found.
/home/u/MS/recommenders/recommenders/utils/tf_utils.py:docstring of recommenders.utils.tf_utils.evaluation_log_hook:26: WARNING: Inline strong start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/u/MS/recommenders/docs/index.rst: WARNING: document isn't included in any toctree
done
```

**Comment by miguelgfierro** [comment]
The error: `ERROR: Unexpected indentation.` means that there is no docstring o_o

Very descriptive. Congrats!!

**Comment by miguelgfierro** [comment]
Error. The documentation is not rendered correctly, there are extra : 
image

**Comment by miguelgfierro** [comment]
Error when trying to push the files from the GitHub actions to the repo:
```
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.
```

**Comment by miguelgfierro** [comment]
In PyTorch, they are using sphinx directly: 
- [conf.py](https://github.com/pytorch/pytorch/blob/main/docs/source/conf.py)
- [make](https://github.com/pytorch/pytorch/blob/main/docs/Makefile)

**Comment by agoose77** [comment]
@miguelgfierro could you try with the newest version of JB?

**Comment by miguelgfierro** [comment]
@agoose77 it works now https://recommenders-team.github.io/recommenders/

Thanks man!

File: docs/_config.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# Book settings
+# Learn more at https://jupyterbook.org/customize/config.html
+
+# To build the Jupyter Book:
+# $ jupyter-book clean docs 
+# $ jupyter-book build docs
+
+
+title: Recommenders documentation
+author: Recommenders contributors
```
Comment (SimonYansenZhao) [inline docs/_config.yml:13]:
```suggestion
author: Recommenders contributors
copyright: "2018-2024"
```

Comment (SimonYansenZhao) [reply docs/_config.yml:13]: If the copyright is not added, the default copyright will be "2022" as I test.

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+html:
+  favicon                     : https://raw.githubusercontent.com/recommenders-team/artwork/main/icon/recommenders_color_icon.svg
+  home_page_in_navbar         : false
+  use_repository_button       : true
+  use_issues_button           : true
+  baseurl                     : https://recommenders-team.github.io/recommenders/
+
+sphinx:
+  extra_extensions:
+  - sphinx_inline_tabs
```
Comment (SimonYansenZhao) [inline docs/_config.yml:46]: I don't find any inline tabs (`.. tab::`) in the docs, so `sphinx_inline_tabs` is not used.

```suggestion
```

File: docs/requirements-doc.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+jupyter-book>=1.0.0
+sphinx>5.0.0
+sphinx_inline_tabs
```
Comment (SimonYansenZhao) [inline docs/requirements-doc.txt:3]: `sphinx` is required by `Jupyter-book`, and `sphinx_inline_tabs` is not used, so they can be removed.

```suggestion
```

File: docs/requirements-doc.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+ghp-import
```
Comment (SimonYansenZhao) [inline docs/requirements-doc.txt:4]: I don't find any usage of `ghp-import` either.

**Review by SimonYansenZhao — APPROVED** [review]

**Comment by SimonYansenZhao** [comment]
[.readthedocs.yaml](https://github.com/recommenders-team/recommenders/blob/main/.readthedocs.yaml) is not used any more if we use Jupyter Book.  @miguelgfierro

File: docs/requirements-doc.txt (hunk 4 above)
Comment (miguelgfierro) [reply docs/requirements-doc.txt]: added here: https://github.com/recommenders-team/recommenders/pull/2051/commits/1aa1945d2d3190e053fb47c529fc58db7ebe3a65

**Pull request merged**